File: cfgs/java_cfgs_foo/ZECKENDORFS_THEOREM_NON_NEIGHBOURING_FIBONACCI_REPRESENTATION.java
Nodes
1;special;;;
2;conditional;(foo_1 == 0) || (foo_2 == 1);14;
3;statement;return foo_1;14;
4;exit;;;
5;statement;int f1 = 0;15;
6;statement;int f2 = 1;15;
7;statement;int f3 = 1;15;
8;conditional;foo_1 <= foo_2;16;
9;statement;f1 = foo_1;17;
10;statement;f2 = foo_1;18;
11;statement;f3 = foo_1 + foo_2;19;
12;statement;return foo_1;21;
Edges
0;2;
1;3;5;
2;4;
3;
4;6;
5;7;
6;8;
7;12;9;
8;10;
9;11;
10;8;
11;4;

File: cfgs/pythoncfgs/ZECKENDORFS_THEOREM_NON_NEIGHBOURING_FIBONACCI_REPRESENTATION.py
Nodes
1;conditional;foo_1 == 0 or foo_2 == 1;8;
2;statement;(foo_1, foo_2, foo_3) = (0, 1, 1);10;
3;conditional;foo_1 <= foo_2;11;
4;statement;return foo_1;15;
5;statement;foo_1 = foo_2;12;
6;statement;foo_1 = foo_2;13;
7;statement;foo_1 = foo_2 + foo_3;14;
8;statement;return foo_1;9;
9;exit;;
Edges
0;2;8;
1;3;
2;4;5;
3;9;
4;6;
5;7;
6;3;
7;9;
8;